### ISSUE TITLE
Add Wordshoal for multi-domain political text scaling

### ISSUE THREAD

**Issue by nealcaren** [nealcaren/topica#695]
## What you want and why

Add **Wordshoal** (Lauderdale & Herzog, 2016), a multi-domain extension of Wordfish for estimating a shared speaker/party scale from texts grouped into known debate domains. The model fits domain-level word-scaling measurements and combines them into an actor-level latent scale.

This is a useful complement to topica's existing political-text family:

- `Wordfish` estimates a scale for text about one issue/domain.
- `TBIP` learns topics and an ideal point from unlabeled multi-issue text.
- `Wordshoal` is appropriate when documents have trusted, externally supplied debate/agenda domains and the researcher wants comparability across them.

Reference: Benjamin E. Lauderdale and Alexander Herzog (2016), *Measuring Political Positions from Legislative Speech*, Political Analysis 24(3):374–394. https://doi.org/10.1093/pan/mpw017

## Proposed surface

```python
model = topica.Wordshoal(seed=13)
model.fit(docs, speakers=speaker_id, domains=debate_id)

model.actor_positions
model.domain_scales
model.word_scores(domain="health")
topica.viz.ideal_points(model)
```

`domains` should be categorical, externally observed labels—not inferred topics. Provide stable identifiability conventions and explicit orientation controls, as for `Wordfish`.

## Acceptance criteria

- Fits domain-specific Wordfish-style measurements and an actor-level aggregation model.
- Supports repeated speakers across domains and documents with one speaker/domain label each.
- Documents data requirements: adequate within-domain vocabulary, overlap of speakers across domains, and sensitivity to preprocessing/domain definitions.
- Exposes actor positions, uncertainty where the model supports it, domain diagnostics, and interpretable word-level contributions.
- Includes orientation/identification controls and save/load support.
- Validation includes simulation recovery, comparison against the paper or available reference implementation, and failure cases for disconnected speaker-domain designs.

## Alternatives

Use `TBIP` when issue domains are unknown or should be learned from unstructured text; it does not require supplied debate labels. Use `Wordfish` for a single known issue. Wordshoal should not be presented as a general replacement for either.
